src/app/components/Search/TableSearchBar.tsx

```diff
@@ -0,0 +1,135 @@
+import { FC, useEffect, useState } from 'react'
+import TextField from '@mui/material/TextField'
+import SearchIcon from '@mui/icons-material/Search'
+import HighlightOffIcon from '@mui/icons-material/HighlightOff'
+import InputAdornment from '@mui/material/InputAdornment'
+import { COLORS } from '../../../styles/theme/colors'
+import IconButton from '@mui/material/IconButton'
+import { useScreenSize } from '../../hooks/useScreensize'
+import WarningIcon from '@mui/icons-material/WarningAmber'
+import { typingDelay } from '../../../styles/theme'
+import Typography from '@mui/material/Typography'
+import { useTranslation } from 'react-i18next'
+import Button from '@mui/material/Button'
+import { CardEmptyState } from '../CardEmptyState'
+
+export interface TableSearchBarProps {
+  placeholder: string
+  warning?: string
+  error?: string
+  value: string
+  onChange: (value: string) => void
+}
+
+export const TableSearchBar: FC<TableSearchBarProps> = ({ value, onChange, placeholder, error, warning }) => {
+  const { isTablet } = useScreenSize()
+
+  const [isProblemFresh, setIsProblemFresh] = useState(false)
+  const hasProblem = !!error || !!warning
```

Comment:
Maybe it is just me, but I don't like `Problem` wording. Maybe `issue` or maybe even `exception`.

Reply:
Since we are not using the error prop for everything, I have just removed it, and then we don't even need this aggregated problem flag, because we can just check `!!warning`. So, removed it.

Reply:
`const [isProblemFresh, setIsProblemFresh] = useState(false)`

Reply:
Renamed to use warning instead of problem.